### PR TITLE
Update scrapertest package to include the ability to ignore values

### PR DESCRIPTION
**Pull request by Mrod1598**
**Description:** <Describe what has changed.>
Added flag to compare functions to ignore comparing the values of a datapoint slice. This will enable these functions to be used within integration tests where we don't necessarily know what the values will be.

### CODE REVIEW

**Review by djaglowski — COMMENTED** [review]
Looks good, w/ a few nits. 

The circle check is being looked at separately.

File: internal/scrapertest/scrapertest.go [hunk 1]
```diff
@@ -154,15 +154,17 @@ func CompareNumberDataPointSlices(actual, expected pdata.NumberDataPointSlice) e
...
-	if expected.DoubleVal() != actual.DoubleVal() {
-		return fmt.Errorf("metric datapoint DoubleVal doesn't match expected: %f, actual: %f", expected.DoubleVal(), actual.DoubleVal())
+	if checkValues {
+		if expected.IntVal() != actual.IntVal() {
+			return fmt.Errorf("metric datapoint IntVal doesn't match expected: %d, actual: %d", expected.IntVal(), actual.IntVal())
+		}
+		if expected.DoubleVal() != actual.DoubleVal() {
+			return fmt.Errorf("metric datapoint DoubleVal doesn't match expected: %f, actual: %f", expected.DoubleVal(), actual.DoubleVal())
+		}
 	}
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply internal/scrapertest/scrapertest.go]: updated

File: internal/scrapertest/scrapertest_test.go [hunk 2]
```diff
@@ -338,7 +338,52 @@ func TestCompareMetrics(t *testing.T) {
...
+
+func TestCompareMetricsWithoutComparingValues(t *testing.T) {
+	tcs := []struct {
+		name          string
+		expected      pdata.MetricSlice
+		actual        pdata.MetricSlice
+		expectedError error
+	}{
+		{
+			name: "Wrong doubleVal",
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply internal/scrapertest/scrapertest_test.go]: updated

File: internal/scrapertest/scrapertest_test.go [hunk 3]
```diff
@@ -338,7 +338,52 @@ func TestCompareMetrics(t *testing.T) {
...
+				m := metrics.At(0)
+				dp := m.Gauge().DataPoints().At(0)
+				dp.SetDoubleVal(123)
+				return metrics
+			}(),
+			expected:      baseTestMetrics(),
+			expectedError: nil,
+		},
+		{
+			name: "Wrong intVal",
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply internal/scrapertest/scrapertest_test.go]: updated

**Review by codeboten — COMMENTED** [review]
I'm not sure I understand the use-case for this, do you have an example of a test where you would want to use these functions without checking the equivalence of the values being compared?

**Comment by djaglowski** [comment]
@codeboten The [nginxreceiver's integration_test.go](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/nginxreceiver/integration_test.go#L100:L142) is a good example of where this would be useful. The scraper is run against a containerized instance of the target application, and so the actual values are not predictable. 

Specifically, the current level of validation in that test is really just a poor approximation of what this functionality can replace. I expect that we'll add similar "integration" tests for dozens of other scrapers and that this pattern can allow for maximum validation with minimal effort.

@codeboten Do you think this PR should include updating the nginxreceiver integration test to make use of this functionality?

**Comment by djaglowski** [comment]
@codeboten The [nginxreceiver's integration_test.go](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/nginxreceiver/integration_test.go#L100:L142) is a good example of where this would be useful. The scraper is run against a containerized instance of the target application, and so the actual values are not predictable. 

Specifically, the current level of validation in that test is really just a poor approximation of what this functionality can replace. I expect that we'll add similar "integration" tests for dozens of other scrapers and that this pattern can allow for maximum validation with minimal effort.

@codeboten Do you think this PR should include updating the nginxreceiver integration test to make use of this functionality?

**Comment by Mrod1598** [comment]
@codeboten Another good example is the memcached receiver's [integration test](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/memcachedreceiver/integration_test.go). All we're currently doing is testing for a name because the values are unpredictable with the container.  where we could use this functionality to test a lot more.

**Comment by codeboten** [comment]
Thanks for the examples @Mrod1598 & @djaglowski, I think we can move with this PR as is, and update the integration tests in a followup PR.

**Review by codeboten — APPROVED** [review]

**Comment by codeboten** [comment]
@djaglowski please approve if your comments have been addressed.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**